### PR TITLE
Remove context pointer

### PR DESCRIPTION
**Pull request by jeffcharles**
This is an example of how we could remove the context pointer from all the APIs. It simplifies the API for the consumer since they no longer need to be aware of the context, there's no pointer they need to store and pass. It also reduces fuel usage by ~6% in the tests because we do not need to perform null checks on the context.

My plan is to build on this to remove `shopify_function_context_new` and `shopify_function_output_finalize` from the partner facing API and have the host environment perform the equivalent operations. Having the host perform these operations reduces the scope of what partner is responsible for (these operations should always be the same for all Functions) and will give us the option to exclude these operations from the fuel and timeout limits given these aren't really partner code and are required for a Function to be useful.

### CODE REVIEW

File: api/src/lib.rs [hunk 1]
```diff
@@ -244,62 +186,54 @@ impl InternedStringId {
 pub struct CachedInternedStringId {
     value: &'static str,
     #[cfg(target_pointer_width = "32")]
-    interned_string_id_and_context: AtomicU64,
+    interned_string_id: AtomicU32,
     #[cfg(target_pointer_width = "64")]
-    interned_string_id_and_context: Mutex<(usize, usize)>,
+    interned_string_id: AtomicU64,
```
Comment (adampetro) [inline api/src/lib.rs:191]: Can we just use `AtomicUsize` and remove the conditional compilation?

Comment (jeffcharles) [reply api/src/lib.rs]: Good call!

**Review by saulecabrera — APPROVED** [review]
LGTM
